### DISCUSSION TITLE
When are objects freed through auto_free

### DISCUSSION THREAD

**Discussion by derkork** [MikeSchulze/gdUnit3#146] [FAQ] [answered]
When you queue an object for being automatically freed using the  `auto_free` function, when will the object actually be freed? Lets assume the following code:

```gdscript
class Test
extends GdUnitTestSuite

var some_shared_Node:Node
var some_per_test_node:Node

func before():
    some_shared_node = auto_free(Foo.new())

func before_test():
    some_per_test_node = auto_free(Bar.new())


```

Now question is, when will `some_shared_node` and `some_per_test_node` be freed? Will it happen after each test execution or only after the execution of the whole suite?

**Comment by MikeSchulze** [comment, marked as answer]
Hi derkork,
Objects covered by `auto_free` live only in the scope where it is used.

Scopes are:
 - test suite setup
   -  starts with `before()` and ends with `after()`
   - finally after `after()` is called the memory fill be auto freed.
- test setup
  - starts with `before_test()` and ends with `after_test()`
  - finally after `after_test()` is called the memory fill be auto freed.
- test
  - lives only inside test()`
  - finally after `test()` is called the memory fill be auto freed.

If this anwsers your question?
I will next improve the documentation to add the missing info.

**Reply by derkork** [reply]
Yep that answers the question. It is actually a very neat way of handling this. Thank you!

**Reply by MikeSchulze** [reply]
Here the updated doc https://github.com/MikeSchulze/gdUnit3/wiki/Tools#a-small-example-test-suite-to-show-up-memory-usage-and-freeing
If you have additional suggestions feel free to inform me. ;)

**Reply by derkork** [reply]
That is a very nice write-up, thank you very much!
